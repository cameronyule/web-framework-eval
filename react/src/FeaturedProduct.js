import React from 'react';

const FeaturedProduct = (props) => (
  <div className="featuredproduct">
    <a href={props.href}>
      <p>{props.title}</p>
      <p>{props.description}</p>
    </a>
  </div>
);

FeaturedProduct.propTypes = {
  href: React.PropTypes.string.isRequired,
  title: React.PropTypes.string.isRequired,
  description: React.PropTypes.string.isRequired
}

export default FeaturedProduct;
