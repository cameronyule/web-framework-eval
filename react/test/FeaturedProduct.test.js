import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import FeaturedProduct from 'FeaturedProduct';

describe('FeaturedProduct', () => {
  let validElement;

  beforeEach(() => {
    validElement = <FeaturedProduct href='#'
                                    title='Test'
                                    description='A Test' />;
  })

  it('renders correctly', () => {
    const wrapper = shallow(validElement);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('has a valid class', () => {
    const wrapper = shallow(validElement);
     expect(wrapper.at(0).prop('className')).toEqual('featuredproduct');
  });

  it('requires an href', () => {
    console.error = jest.fn();
    shallow(<FeaturedProduct title='Test'
                             description='A Test' />);
    expect(console.error).toBeCalled();
    expect(console.error.mock.calls[0][0]).toContain('href');
  });

  it('accepts an href', () => {
    const wrapper = shallow(validElement);
    expect(wrapper.find('a').prop('href')).toEqual('#');
  });

  it('requires a title', () => {
    console.error = jest.fn();
    shallow(<FeaturedProduct href='#'
                             description='A Test' />);
    expect(console.error).toBeCalled();
    expect(console.error.mock.calls[0][0]).toContain('title');
  });

  it('accepts a title', () => {
    const wrapper = shallow(validElement);
    expect(wrapper.find('p').first().text()).toEqual('Test');
  });

  it('requires a description', () => {
    console.error = jest.fn();
    shallow(<FeaturedProduct href='#'
                             title='Test' />);
    expect(console.error).toBeCalled();
    expect(console.error.mock.calls[0][0]).toContain('description');
  });

  it('accepts a description', () => {
    const wrapper = shallow(validElement);
    expect(wrapper.find('p').at(1).text()).toEqual('A Test');
  });
})
