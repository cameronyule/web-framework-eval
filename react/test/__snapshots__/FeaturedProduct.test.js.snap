exports[`FeaturedProduct renders correctly 1`] = `
<div
  className="featuredproduct">
  <a
    href="#">
    <p>
      Test
    </p>
    <p>
      A Test
    </p>
  </a>
</div>
`;
